bin/internal/content_aware_hash.sh

```diff
@@ -44,7 +44,7 @@ if [[ "$CURRENT_BRANCH" != "main" && \
       "$CURRENT_BRANCH" != "beta" && \
       "$CURRENT_BRANCH" != "gh-readonly-queue/master/pr-"* && \
       "$CURRENT_BRANCH" != "flutter-"*"-candidate."* && \
-      "$CURRENT_BRANCH" != "HEAD" && \
+      ! ( "$CURRENT_BRANCH" == "HEAD" && -n "$LUCI_CI" ) && \
```

Comment:
ok, I checked out HEAD~ to get into detached head (while on master) just to see if this would still compute. tl/dr it does.  Why this is important: people still need to do bisect.

What was the problem with LUCI_CI?

Reply:
The problem arises when you are in a detached HEAD state and also have branched off of `main`. Being in detached HEAD but still on a commit in main will work because there will be a Dart SDK available to download with that content hash. But if you are in detached HEAD and also branched off main, then there is no Dart SDK uploaded for that content hash and so downloading the Dart SDK will fail. I assume it works on LUCI because the builders put themselves in a detached HEAD state by checking out the contents of a PR, but in this case a Dart SDK for that content hash has already been uploaded in an earlier builder step.

Reply:
Pretty please keep an eye on this when it lands.